src/key.rs

```diff
@@ -1,36 +1,35 @@
+use core::fmt;
+
 use crate::{cycle::CycleRecoveryStrategy, zalsa::IngredientIndex, Database, Id};
 
 /// An integer that uniquely identifies a particular query instance within the
-/// database. Used to track dependencies between queries. Fully ordered and
+/// database. Used to track output dependencies between queries. Fully ordered and
 /// equatable but those orderings are arbitrary, and meant to be used only for
 /// inserting into maps and the like.
 #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct DependencyIndex {
-    pub(crate) ingredient_index: IngredientIndex,
-    pub(crate) key_index: Option<Id>,
+pub struct OutputDependencyIndex {
```

Comment:
Do we rely on the `Ord` implementation anywhere? I suspect not because `QueryEdge` doesn't implement `Ord`. Should we, therefore, remove it?

Reply:
That is a good point to raise, I agree I'll remove them.

Reply:
I went a head and removed it from `DatabaseKeyIndex` as well (and hence `Cycle`), as it was only used in one place and there the ord call won't ever be reached (as its behind the ordering of an enumerator in a tuple), [`189fc09` (#644)](https://github.com/salsa-rs/salsa/pull/644/commits/189fc09055896f3a6306374532cdafc48a5990a5)